tests/scripts/generate_test_code.py

```diff
@@ -255,6 +255,7 @@ def get_line_no(self):
 def split_dep(dep):
     """
     Split NOT character '!' from dependency. Used by gen_dependencies()
+    Determine condition MACRO and definition MACRO.
```

Comment:
Since the Python code just translates the dependencies to C preprocessor directives, why not allow arbitrary code (`[^\n:]+`)?

Reply:
This line is a residue from when I tried first determining the conditions to definitions.
This function basically returns the `!` operator

Reply:
(continued [here](https://github.com/ARMmbed/mbedtls/pull/2231#discussion_r236806794))